### ISSUE TITLE
BUG: np.typeDict['Complex32'] is np.compex64

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#10150]
This is just super confusing:

```python
>>> np.complex64
<type 'numpy.complex64'>
>>> np.typeDict['Complex32']
<type 'numpy.complex64'>
>>> np.typeNA['c16']
<type 'numpy.complex64'>
```

**Comment by charris** [comment]
Left over type from numarray. We should really get rid of all of them at some point, but given as we cannot deprecate them, we will probably have to just do it an see who yells.

**Comment by eric-wieser** [comment]
Should be possible to deprecate them by subclassing `dict`. We already deprecate these elsewhere in #7743 though, so it's maybe safe to just drop them entirely

**Comment by eric-wieser** [comment]
Closed in #16554
